services/app-api/src/domain-models/contractAndRates/formDataTypes.ts

```diff
@@ -34,40 +35,55 @@ const packagesWithSharedRateCerts = z.object({
     packageStatus: statusSchema.optional(),
 })
 
+// Wraps schema to make it optional and transforms null to undefined. This will take nulls from graphql inputs and turn them to undefined.
+const nullishTransformWrapper = (schema: ZodSchema) =>
```

Comment:
I've updated this to use `z.preprocess()`. The type issue we saw was because I defined the function argument types incorrectly.

```ts
function preprocessNulls<T extends ZodTypeAny>(
    schema: T,
    params?: RawCreateParams
): ZodEffects<T, T['_output'], unknown> {
    return z.preprocess((val) => val ?? undefined, schema, params)
}
```
I think this could replace converter functions like `contractFormDataToDomainModel()`. Something like `prismaUpdateContractFormDataFromDomain` I'm not too sure about, the `safeParse` output type would be more of a Prisma schema.